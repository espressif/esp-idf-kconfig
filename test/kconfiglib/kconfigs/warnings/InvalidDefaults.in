mainmenu "Invalid default values"

comment "There is no invalid value for string type -> it is omitted here"
# Valid configs for reference

config INT_CONFIG
    int "Int config"
    default 1

config BOOL_CONFIG
    bool "Bool config"
    default y

config HEX_CONFIG
    hex "Hex config"
    default 0x42


# Invalid direct defaults

# Kconfig quirk: although string is not a valid int (and it is recognized as invalid),
# value will be set nevertheless
config INVALID_INT_CONFIG
    int "Invalid Int config"
    default "string"

config INVALID_BOOL_CONFIG
    bool "Invalid Bool config"
    default 0x42

# Kconfig quirk: although y is not a valid hex (and it is recognized as invalid),
# value will be set nevertheless
config INVALID_HEX_CONFIG
    hex "Invalid Hex config"
    default y

config INT_OUTSIDE_RANGE_CONFIG
    int "Outside range config"
    default 1000
    range 0 999

config HEX_OUTSIDE_RANGE_CONFIG
    hex "Hex outside range config"
    default 0x100
    range 0x0 0xFF


# Invalid indirect defaults

config SOURCE
    bool "Source config"
    default y
    set INT_CONFIG="string"
    set BOOL_CONFIG=0x42
    set HEX_CONFIG="y"

# Invalid weak indirect defaults

config WEAK_SOURCE
    bool "Weak source config"
    default y
    set INT_CONFIG="weak"
    set BOOL_CONFIG=0x43
    set HEX_CONFIG=220
