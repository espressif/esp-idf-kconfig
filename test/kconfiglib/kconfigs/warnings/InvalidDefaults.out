#
# There is no invalid value for string type -> it is omitted here
#
# default:
CONFIG_INT_CONFIG=1
# default:
CONFIG_BOOL_CONFIG=y
# default:
CONFIG_HEX_CONFIG=0x42
# default:
CONFIG_INVALID_INT_CONFIG=string
# default:
# CONFIG_INVALID_BOOL_CONFIG is not set
# default:
CONFIG_INVALID_HEX_CONFIG=y
# default:
CONFIG_INT_OUTSIDE_RANGE_CONFIG=999
# default:
CONFIG_HEX_OUTSIDE_RANGE_CONFIG=0xff
# default:
CONFIG_SOURCE=y
# default:
CONFIG_WEAK_SOURCE=y
